packages/router/router.ts

```diff
@@ -752,8 +780,9 @@ export function createRouter(init: RouterInit): Router {
     }
 
     // Call action if we received an action submission
-    let pendingActionData: RouteData | null = null;
-    let pendingError: RouteData | null = null;
+    let request = createRequest(location, opts?.submission);
```

Comment:
This is the big change to current router internals. In order to better align the server/browser use-cases we lift `Request` creation way up from where it used to be. That way we're better able to accept an external `Request` in the SSR case